### DISCUSSION TITLE
How make a variable exactly same with number of row when using repeater component ?

### DISCUSSION THREAD

**Discussion by jigujigu** [filamentphp/filament#3950] [Help]
Hi All,

How I can make my value in Line# column exactly same with  number of row when using repeater component ?
and how I can re-order the Line# if one of row is deleted ?

Thank You

```
    \Filament\Forms\Components\Repeater::make('orderdetails')
        ->label('Item Line')
        ->columnSpan('full')
        ->relationship()
        ->columns(12)
        ->schema([

            \Filament\Forms\Components\TextInput::make('line_number')
                ->mask(
                    fn (\Filament\Forms\Components\TextInput\Mask $mask) => $mask
                        ->numeric()->integer())
                ->minLength(1)->maxLength(2)
                ->label('Line#')
                ->default(fn () => 1)  //    <---- what function to return the row number ?

            \Filament\Forms\Components\TextArea::make('description')
                ->minLength(5)
                ->label('Description')
                ->rows(2)
        ]),
```

![image](https://user-images.githubusercontent.com/6214964/189500863-627b682c-7956-432e-a018-d29e3f4908e9.png)

**Comment by danharrin** [comment]
It's possible, but I question as to why you would need to do this? When the data is saved, the repeater will have a true numeric key anyway, and then you dont need to worry about keeping it up to date.

**Comment by borex** [comment]
I'm interested in the same thing. In some cases is good to display the row number in the list and when moving the items up and down the list, also the row number should update. Is there a solution for this or how can it be done?

**Comment by borex** [comment]
I solved the row number problem the following way:

`
...
                    ->registerListeners([
                        "repeater::createItem" => [ function (Repeater $component, string $statePath): void {
                            if ($component->isDisabled()) {
                                return;
                            }

                            if ($statePath !== $component->getStatePath()) {
                                return;
                            }
        
                            $items = $component->getState();
                            $newUuid = array_key_last($items);
                            $items[$newUuid]['display_order'] = count($items);

                            $livewire = $component->getLivewire();
                            data_set($livewire, "{$statePath}.{$newUuid}", $items[$newUuid]);
                        }, ]
                    ])
                    ->schema([
                        Forms\Components\TextInput::make('display_order')
                            ->default(1)
                            ->label(__('Row number')),
                    ])
...
`

I'm pretty new to Filament, so bear with me and if this is not the correct way to do this. If somebody has a better solution I'd like to hear it. Thank you.

**Comment by vuthaihoc** [comment]
I use this code

![image](https://github.com/user-attachments/assets/b348de44-2763-4492-9c21-5fa86decb5be)

Result

![image](https://github.com/user-attachments/assets/6b79179b-cead-4ad2-8d22-be8a7de96e39)

I'm using `awcodes/filament-table-repeater` but i think this code will work with any implementation of `Repeater`
